src/System.Private.CoreLib/shared/System/WeakReference.T.cs

```diff
@@ -52,7 +52,7 @@ internal WeakReference(SerializationInfo info, StreamingContext context)
         //          DoSomething(ref.Target)
         //
         [MethodImplAttribute(MethodImplOptions.AggressiveInlining)]
-        public bool TryGetTarget([NotNullWhen(true)] out T target)
+        public bool TryGetTarget([MaybeNull, NotNullWhen(true)] out T target)
```

Comment:
doesn't NotNullWhen imply MaybeNull? If not should one derive from another?

Reply:
I'd thought it did, but was corrected by @MadsTorgersen.

Reply:
This makes me sad, I think the compiler should accept T? even if the constraint it : class?.

To me the constraint means "what can I use to create an instance of this class" because if I put the constraint as `: class` then I can't create an instance of `Foo<string?>` but then if I put a constraint of `: class?` I can but then I need to specify where a return or an out parameter `MaybeNull` 😢